test/integration/features/addons/addon-registry-route.feature

```diff
@@ -18,21 +18,34 @@ so that user can able to push their container images directly to registry and de
       Add-on 'registry-route' created docker-registry route.
       """
 
-  Scenario: As a user, I can login to OpenShift integrated docker registry
-    Given Minishift has state "Running"
-      And executing "minishift docker-env" in host shell succeeds
-      And evaluating stdout of the previous command in host shell
-      And executing "minishift oc-env" in host shell succeeds
-      And evaluating stdout of the previous command in host shell
+  Scenario: User waits until registry is fully started and accessible
      Then with up to "20" retries with wait period of "3" seconds container name "registry_docker-registry" should be "running"
-      And with up to "60" retries with wait period of "1" second command "curl -ik https://docker-registry-default.$(minishift ip).nip.io" output should contain "200 OK"
-      And executing "docker login -u developer -p `oc whoami -t` docker-registry-default.$(minishift ip).nip.io" in host shell succeeds
-      And stdout of host shell should contain "Login Succeeded"
+      And with up to "60" retries with wait period of "1000ms" the "status code" of HTTP request to "/" of service "docker-registry" in namespace "default" is equal to "200"
+
+  Scenario: User sets docker environment variables
+     When executing "minishift docker-env" in host shell succeeds
+      And evaluating stdout of the previous command in host shell succeeds
```

Comment:
Can this be `evaluating stdout of "minishift docker-env" command in host shell succeeds" instead of two statements i.e. 
```
When executing "minishift docker-env" in host shell succeeds
    And evaluating stdout of the previous command in host shell succeeds
```

Reply:
Forget that the second statement should be `Then...`. I would not combine those into 1 step as this could make it much harder to understand in case of error. Does the command failed, or does the evaluation failed? We can get into logs, but that is not ideal sometimes.